### DISCUSSION TITLE
Error: Cannot find module '@azure/ms-rest-nodeauth' with 2.10.12

### DISCUSSION THREAD

**Discussion by lanedsmu** [Budibase/budibase#11846] [Help] [answered]
Hi, in trying to update to the latest version, we get a Node "cannot find module" error with Docker (full error below).  Any ideas on how to work around or fix this?  We /do/ use AzureAD for our BB authentication.  

I've pulled bbapps back to 2.8.22, and I still get this error. 

Thanks!  -Lane
 --
bbapps                | node:internal/modules/cjs/loader:1080
bbapps                |   throw err;
bbapps                |   ^
bbapps                |
bbapps                | Error: Cannot find module '@azure/ms-rest-nodeauth'
bbapps                | Require stack:
bbapps                | - /app/node_modules/tedious/lib/connection.js
bbapps                | - /app/node_modules/tedious/lib/tedious.js
bbapps                | - /app/node_modules/mssql/lib/tedious/connection-pool.js
bbapps                | - /app/node_modules/mssql/lib/tedious/index.js
bbapps                | - /app/node_modules/mssql/index.js
bbapps                | - /app/dist/index.js
bbapps                |     at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
bbapps                |     at Module._load (node:internal/modules/cjs/loader:922:27)
bbapps                |     at Module.require (node:internal/modules/cjs/loader:1143:19)
bbapps                |     at require (node:internal/modules/cjs/helpers:119:18)
bbapps                |     at Object.<anonymous> (/app/node_modules/tedious/lib/connection.js:16:23)
bbapps                |     at Module._compile (node:internal/modules/cjs/loader:1256:14)
bbapps                |     at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
bbapps                |     at Module.load (node:internal/modules/cjs/loader:1119:32)
bbapps                |     at Module._load (node:internal/modules/cjs/loader:960:12)
bbapps                |     at Module.require (node:internal/modules/cjs/loader:1143:19)
bbapps                |     at require (node:internal/modules/cjs/helpers:119:18)
bbapps                |     at Object.<anonymous> (/app/node_modules/tedious/lib/tedious.js:59:42)
bbapps                |     at Module._compile (node:internal/modules/cjs/loader:1256:14)
bbapps                |     at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
bbapps                |     at Module.load (node:internal/modules/cjs/loader:1119:32)
bbapps                |     at Module._load (node:internal/modules/cjs/loader:960:12) {
bbapps                |   code: 'MODULE_NOT_FOUND',
bbapps                |   requireStack: [
bbapps                |     '/app/node_modules/tedious/lib/connection.js',
bbapps                |     '/app/node_modules/tedious/lib/tedious.js',
bbapps                |     '/app/node_modules/mssql/lib/tedious/connection-pool.js',
bbapps                |     '/app/node_modules/mssql/lib/tedious/index.js',
bbapps                |     '/app/node_modules/mssql/index.js',
bbapps                |     '/app/dist/index.js'
bbapps                |   ]
bbapps                | }
bbapps                |
bbapps                | Node.js v18.18.0
bbapps                | error Command failed with exit code 1.

**Comment by lanedsmu** [comment]
I can confirm this error doesn't occur when running v2.6.28.  We'll stepwise update and see where the error starts to appear.  Any hints would be appreciated.

**Comment by melohagan** [comment]
Hey @lanedsmu How are you hosting Budibase? I.e. single image, docker-compose, Budi CLI, etc ?

**Reply by lanedsmu** [reply]
we're using docker-compose, with minimal (but some) changes to the docker-compose.yaml.  Basically proxy settings.

**Comment by lanedsmu** [comment]
We’re running v10.24.0

**Reply by melohagan** [reply, marked as answer]
Is it possible to use node version 18?

**Reply by lanedsmu** [reply]
That did it; thanks much!
